src/api/conflictResolution.ts

```diff
@@ -47,98 +62,242 @@ export const getConflictFreeDocuments = ({
-          const rightRepoDoc = asMaybe(asStoreRepoDocument)(rightDoc)
-          const rightDirectoryDoc = asMaybe(asStoreDirectoryDocument)(rightDoc)
-          const rightDirectoryLikeDoc = rightRepoDoc ?? rightDirectoryDoc
-
-          // Merge directory-like documents
-          if (leftDirectoryLikeDoc != null && rightDirectoryLikeDoc != null) {
-            return mergeDirectoryLikeDoc(
-              leftDirectoryLikeDoc,
-              rightDirectoryLikeDoc
-            )
-          }
+    const docs = resultDocs.map(({ ok: doc }) => doc)
 
-          // Documents aren't of the same type
-          if (rightFileDoc != null && leftDirectoryLikeDoc != null) {
-            return sortTimestampedDoc(rightFileDoc, leftDirectoryLikeDoc)[1]
-          }
-          if (leftFileDoc != null && rightDirectoryLikeDoc != null) {
-            return sortTimestampedDoc(leftFileDoc, rightDirectoryLikeDoc)[1]
-          }
+    // Merge the documents
+    const doc = docs.slice(1).reduce<StoreDocument>((leftDoc, rightDoc) => {
+      const leftFileDoc = asMaybe(asStoreFileDocument)(leftDoc)
+      const rightFileDoc = asMaybe(asStoreFileDocument)(rightDoc)
 
-          // Unknown document types:
-          throw new Error(
-            'Unable to merge conlficts for documents of unknown type.'
-          )
-        }
+      // Merge file documents
+      if (leftFileDoc != null && rightFileDoc != null) {
+        const mergedDoc = mergeTimestampedDocs(leftFileDoc, rightFileDoc)
+
+        return mergedDoc
+      }
+
+      const leftRepoDoc = asMaybe(asStoreRepoDocument)(leftDoc)
+      const leftDirectoryDoc = asMaybe(asStoreDirectoryDocument)(leftDoc)
+      const leftDirectoryLikeDoc = leftRepoDoc ?? leftDirectoryDoc
+      const rightRepoDoc = asMaybe(asStoreRepoDocument)(rightDoc)
+      const rightDirectoryDoc = asMaybe(asStoreDirectoryDocument)(rightDoc)
+      const rightDirectoryLikeDoc = rightRepoDoc ?? rightDirectoryDoc
+
+      // Merge directory-like documents
+      if (leftDirectoryLikeDoc != null && rightDirectoryLikeDoc != null) {
+        return mergeDirectoryLikeDocs(
+          leftDirectoryLikeDoc,
+          rightDirectoryLikeDoc,
+          mergeBaseTimestamp
+        )
+      }
+
+      // Documents aren't of the same type
+      if (rightFileDoc != null && leftDirectoryLikeDoc != null) {
+        return mergeTimestampedDocs(rightFileDoc, leftDirectoryLikeDoc)
+      }
+      if (leftFileDoc != null && rightDirectoryLikeDoc != null) {
+        return mergeTimestampedDocs(leftFileDoc, rightDirectoryLikeDoc)
+      }
+
+      // Ignore any document conflicts that were deleted
+      if ('_deleted' in leftDoc) {
+        if (rightFileDoc != null) return rightFileDoc
+        if (rightDirectoryLikeDoc != null) return rightDirectoryLikeDoc
+      }
+      if ('_deleted' in rightDoc) {
+        if (leftFileDoc != null) return leftFileDoc
+        if (leftDirectoryLikeDoc != null) return leftDirectoryLikeDoc
+      }
```

Comment:
Discovery on this: `_bulk_get` will always return the `_deleted` docs and these documents are retained even after compaction. This means we should always handle the case that a document may be a _deleted document.

Furthermore, this impacts performance because even after writing on top of a merged document (resolve) the _deleted documents will be compared to the most recent revision.

Reply:
Add a case here if both documents are deleted, then return one of them.

Also, add to the algorithm a query for documents using `_all_docs?conflicts=true` to first get the conflict free documents from the database and check for any conflict revs in the "conflicts" array returned. If there are conflicts, either use _bulk_get or _all_docs to get the conflicting documents and continue with the resolution algorithm.